### PR TITLE
Release 3.3.0

### PR DESCRIPTION
**Pull request by bording**
Connects to #150

### CODE REVIEW

**Comment by bording** [comment]
@Particular/rabbitmq-transport-maintainers
 
For now, I've left the `QueueingBasicConsumer` in place, so we are getting deprecation warnings. Do we want to keep this as is or actually go through with swapping it out? At this point I think I'd prefer waiting until the class is actually removed.

I also have only bumped the min nuspec range and left the max alone. I think I prefer this approach, but what does everyone else think?

**Comment by SimonCropp** [comment]
> I also have only bumped the min nuspec range and left the max alone. I think I prefer this approach, but what does everyone else think?

I still think this is preferable. yes rabbit breaks things in patches, but very rarely does that effect the small surface area of the API we use. So not fixing the patch means we dont need to release in lockstep with rabbit. i think that is an acceptable compromise

File: src/NServiceBus.RabbitMQ/Connection/RabbitMqConnectionFactory.cs [hunk 1]
```diff
@@ -44,10 +44,9 @@ public RabbitMqConnectionFactory(ConnectionConfiguration connectionConfiguration
             };
         }
 
-        public virtual IConnection CreateConnection(string purpose)
+        public virtual IConnection CreateConnection(string connectionName)
```
Comment (adamralph) [inline src/NServiceBus.RabbitMQ/Connection/RabbitMqConnectionFactory.cs]: I know this is an edge case, but this is potentially breaking if someone is naming the parameter (same applies to the change to `PersistentConnection.ctor()`). Do we have a policy on this? Personally I'm happy to call it non-breaking.

**Comment by adamralph** [comment]
:+1: LGTM. What's left to do?

File: src/NServiceBus.RabbitMQ/Connection/RabbitMqConnectionFactory.cs (hunk 1 above)
Comment (SimonCropp) [inline src/NServiceBus.RabbitMQ/Connection/RabbitMqConnectionFactory.cs]: What is the compelling reason to make this changes? Does it justify breaking semver?

Comment (adamralph) [inline src/NServiceBus.RabbitMQ/Connection/RabbitMqConnectionFactory.cs]: [Every public change is a breaking change](https://blogs.msdn.microsoft.com/ericlippert/2012/01/09/every-public-change-is-a-breaking-change/) so there is always a class of theoretical breakages which need to be excluded from a project's definition of breaking, but I second the challenge. @bording what is the reason for making this change?

Comment (bording) [inline src/NServiceBus.RabbitMQ/Connection/RabbitMqConnectionFactory.cs]: Maybe I'm missing something here. `RabbitMqConnectionFactory` isn't public, so this isn't a public-facing change.

Comment (adamralph) [inline src/NServiceBus.RabbitMQ/Connection/RabbitMqConnectionFactory.cs]: > they're both internal classes

Well, yes, that changes things :laughing: 

OK, ignore all of the above. Sorry for derailing. :grin:

**Comment by bording** [comment]
> :+1: LGTM. What's left to do?

At this point, it should be ready to release. Feel free to go ahead and do that, otherwise I'll get it out tomorrow.

**Comment by bording** [comment]
While preparing this release, I found a bug in the new connection name feature introduced in the RabbitMQ 3.6.2 client (https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/187), so I'm going to create a 3.3.1 release and re-add the "purpose" client property to surface the value for now.

**Pull request merged**